dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -321,11 +321,45 @@
 			<!-- Mark our entry assembly as a root assembly. -->
 			<TrimmerRootAssembly Include="@(ResolvedFileToPublish)" Condition="'%(ResolvedFileToPublish.Filename)' == '$(AssemblyName)' And '%(ResolvedFileToPublish.Extension)' == '.dll'" />
 
-			<!-- add a custom step which inserts any other steps we need -->
-			<_TrimmerCustomSteps Include="$(_AdditionalTaskAssembly)">
-				<BeforeStep>MarkStep</BeforeStep>
-				<Type>Xamarin.SetupStep</Type>
-			</_TrimmerCustomSteps>
+			<!-- pre-mark custom steps -->
+			<_TrimmerCustomSteps Include="$(_AdditionalTaskAssembly)" BeforeStep="MarkStep" Type="Xamarin.SetupStep" />
+			<!-- Load the list of assemblies loaded by the linker. -->
+			<!-- This would not be needed of LinkContext.GetAssemblies () was exposed to us. -->
+			<_TrimmerCustomSteps Include="$(_AdditionalTaskAssembly)" BeforeStep="MarkStep" Type="Xamarin.Linker.CollectAssembliesStep" />
+			<!-- TODO: these steps should probably run after mark. -->
+			<_TrimmerCustomSteps Include="$(_AdditionalTaskAssembly)" BeforeStep="MarkStep" Condition="'$(_LinkMode)' != 'None'" Type="Xamarin.Linker.Steps.StoreMembersAndAttributesDispatcher" />
+			<!-- IMarkHandlers which run during Mark -->
+			<_TrimmerCustomSteps Include="$(_AdditionalTaskAssembly)" Condition="'$(_LinkMode)' != 'None'" Type="Xamarin.Linker.Steps.PreserveBlockCodeHandler" />
+			<!--
+				We need to run the ApplyPreserveAttribute step even we're only linking sdk assemblies, because even
+				though we know that sdk assemblies will never have Preserve attributes, user assemblies may have
+				[assembly: LinkSafe] attributes, which means we treat them as sdk assemblies and those may have
+				Preserve attributes.
+				TODO: LinkSafeAttribute doesn't appear to be handled anywhere. Is this attribute still supported?
```

Comment:
yes, but that code might just be (for now) in legacy
not everything was ported to dotnet-linker

Reply:
> builds//mono-ios-sdk-destdir/ios-sources/external/linker/src/tuner/Mono.Tuner/CustomizeActions.cs

The code that handle `[LinkerSafe]` is inside mono (repo) and the legacy SDK uses the source code (for a few files) from the mono package (2020-02) when building `mtouch`.